### PR TITLE
Extract common PlatformView functionality: PlatformViewControllerWidget init

### PR DESCRIPTION
**Pull request by cyanglaz**
## Description

This PR is a part of the effort of extracting the common platform view functionalities.
Included:
* The definition of the public APIs: `PlatformViewControllerWidget`, `PlatformViewController` and `PlatformViewSurface`
* The implementation of ``PlatformViewControllerWidget`'s life cycle (creation, dispose)

not included:
* The rendering and gesture handling by `PlatformViewSurface`
* Focus handling by `PlatformViewControllerWidget`

## Related Issues

https://github.com/flutter/flutter/issues/36779

## Tests

I added the following tests:

group 'platform view'
  'PlatformViewController Widget init'
  'PlatformViewController Widget dispose'

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by dnfield — COMMENTED** [review]
It's a bit hard to review this with the number of TODOs - do we need to land all of this right now with stub implementations, or can we just remove the stub implementations and land them when we're ready?

File: packages/flutter/lib/src/widgets/platform_view.dart [hunk 1]
```diff
@@ -13,6 +13,286 @@ import 'focus_manager.dart';
...
+///             controller: controller,
+///        );
+///       }, onPlatformViewCreated: (int id) {
+///         _onPlatformViewCreated(id);
+///      },
+///    );
+///   }
+/// }
+/// ```
+class PlatformViewControllerWidget extends StatefulWidget {
```
Comment (amirh) [inline packages/flutter/lib/src/widgets/platform_view.dart]: As discussed we want a different name here, we actually don't want that name to be PlatformView as we have other plans for a widget with that name, and as people are unlikely to be using this so we don't want them to pick this widget when they look for PlatformView in the api docs.

One proposal was PlatformViewLink (it "links" the platform view to the widget layer).

**Review by amirh — COMMENTED** [review]
Sending a partial review to not lose my comments as you're planning to restructure this PR series.

**Comment by cyanglaz** [comment]
Closing this PR for now. Going to implement the render object first.

**Pull request closed without merging**